lib/config/migrations/custom/dep-types-migration.ts

```diff
@@ -0,0 +1,27 @@
+import is from '@sindresorhus/is';
+import type { PackageRule } from '../../types';
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class DepTypesMigration extends AbstractMigration {
+  override readonly deprecated = true;
+  override readonly propertyName = 'depTypes';
+
+  override run(value: unknown): void {
+    const packageRules: PackageRule[] = this.get('packageRules') ?? [];
+    if (is.array(value)) {
+      for (const depType of value) {
+        if (is.object(depType) && !is.array(depType)) {
+          const depTypeName = (depType as any).depType;
```

Comment:
can we create / use a concrete type for it, i don't like those any casts. they hide / suppress all type safety.

Reply:
We can use `PackageRule` for this PR and create a new one in another PR because the properties inside depRules are random and are pressent in diff configs. For eg. respectLatest is present in `RenovateSharedConfig` and `FilterConfig` but is is also expected in `PackageRule` so some discussion is needed before assigning a concrete type to it.